### PR TITLE
Add package installing functions

### PR DESCRIPTION
**Pull request by yifanlu**

### CODE REVIEW

File: include/psp2/promoterutil.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+typedef struct ScePromoterUtilityLAUpdate {
+	char titleid[12]; // target app
+	char path[128]; // directory of extracted LA update data
+} ScePromoterUtilityLAUpdate;
+
+int scePromoterUtilityInit(void);
+int scePromoterUtilityExit(void);
+int scePromoterUtilityDeletePkg(void *unk);
+int scePromoterUtilityUpdateLiveArea(ScePromoterUtilityLAUpdate *args);
+int scePromoterUtilityPromotePkg(char *path, int unk);
```
Comment (devnoname120) [inline include/psp2/promoterutil.h]: Shouldn't it be `const char *`?

Comment (yifanlu) [inline include/psp2/promoterutil.h]: Yes

File: include/psp2/sysmodule.h [hunk 2]
```diff
@@ -90,9 +90,17 @@ enum {
...
 
+/* internal module IDs */
+enum {
+	SCE_SYSMODULE_PROMOTER_UTIL         = 0x80000024
+};
+
 int sceSysmoduleLoadModule(SceUInt16 id);
 int sceSysmoduleUnloadModule(SceUInt16 id);
 int sceSysmoduleIsLoaded(SceUInt16 id);
+int sceSysmoduleLoadModuleInternal(SceUInt32 id);
```
Comment (yifanlu) [inline include/psp2/sysmodule.h]: I'm not sure what the comment format is, feel free to add it. The internal ones can accept 32 bit ids.

Comment (devnoname120) [inline include/psp2/sysmodule.h]: Here it the format of the comments : https://github.com/vitasdk/vita-headers/blob/540158d9b47ef8e875bee65d3a8baee4fffe7cbc/include/psp2/ctrl.h#L148-L156

I'll push a commit.

**Comment by devnoname120** [comment]
Here is how to document a new library: https://github.com/vitasdk/vita-headers/pull/101/commits/c790f24278142fcdec70cdb664182809ffe7f450

**Comment by yifanlu** [comment]
Can someone add these ids as well? https://wiki.henkaku.xyz/vita/SceSysmodule#Undocumented_Module_IDs

File: include/psp2/sysmodule.h [hunk 3]
```diff
@@ -90,9 +90,17 @@ enum {
 	SCE_SYSMODULE_AVPLAYER              = 0x004c
 };
 
+/* internal module IDs */
+enum {
```
Comment (devnoname120) [inline include/psp2/sysmodule.h]: Could you change it to a `typedef struct` and give it a name? This is easier for making a reference to it in the documentation.

Also multi-line doxygen-style comments start with `/**` (notice the 2 stars).

**Comment by devnoname120** [comment]
Could you add `scePromoterUtilityRemoveSavedata()` and the promoter heartbeat functinos? https://wiki.henkaku.xyz/vita/ScePromoterUtil

File: include/psp2/promoterutil.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ * @return 0 on success.
+ */
+int scePromoterUtilityExit(void);
+
+/**
+ * Delete a package from the LiveArea.
+ *
+ * @return 0 on success.
+ */
+int scePromoterUtilityDeletePkg(void *unk);
```
Comment (d3m3vilurr) [inline include/psp2/promoterutil.h]: don't know detail info about unk?

Comment (theorywrong) [inline include/psp2/promoterutil.h]: UNKnown, you can try to understand the content of unk

```
 char unk[100]; 
scePromoterUtilityDeletePkg((void*)&unk)
// Save the content of unk
```

File: include/psp2/promoterutil.h [hunk 5]
```diff
@@ -0,0 +1,107 @@
+/**
+ * \usergroup{ScePromoterUtil}
+ * \usage{psp2/promoterutil.h,-lScePromoterUtil_Stub}
+ */
+
+/**
+ * \file
+ * \brief Header file which defines promoter utility related variables and functions
+ *
+ * Copyright (C) 2016 PSP2SDK Project
```
Comment (xyzz) [inline include/psp2/promoterutil.h]: copyright vitasdk

File: include/psp2/promoterutil.h [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (xyzz) [inline include/psp2/promoterutil.h]: can we use MIT instead please?

**Comment by d3m3vilurr** [comment]
well.. only @yifanlu can change license type. just changed argument name and copyright holder.

File: include/psp2/promoterutil.h (hunk 6 above)
Comment (enoposix) [inline include/psp2/promoterutil.h]: New headers can be added as MIT, worry about the others later.

Comment (d3m3vilurr) [inline include/psp2/promoterutil.h]: I make related issue for this. #104

Comment (yifanlu) [inline include/psp2/promoterutil.h]: Changed to MIT

File: include/psp2/promoterutil.h (hunk 5 above)
Comment (yifanlu) [inline include/psp2/promoterutil.h]: Changed

File: include/psp2/promoterutil.h (hunk 4 above)
Comment (yifanlu) [inline include/psp2/promoterutil.h]: nope

Comment (d3m3vilurr) [inline include/psp2/promoterutil.h]: @TheOfficialFloW explained arg meaning, so i was attach commit at yesterday :)

**Pull request merged**